src/geo/transform.js

```diff
@@ -1354,6 +1354,20 @@ class Transform {
         return this.projection.isPointAboveHorizon(this, p);
     }
 
+    /**
+     * Determines if the given point is located on a visible map surface.
+     *
+     * @param {Point} p
+     * @returns {boolean}
+     * @private
+     */
+    isPointOnSurface(p: Point): boolean {
+        if (p.y < 0 || p.y > this.height || p.x < 0 || p.x > this.width) return false;
+        if (this.elevation) return !this.isPointAboveHorizon(p);
+        const coord = this.pointCoordinate(p);
+        return coord.y >= 0 && coord.y <= 1;
```

Comment:
This check works well for the Mercator projection, but I need to figure out how to properly check this for alternate projections.

Reply:
is there a way currently to use this method properly on globe projection?